Lib/test/test_module/__init__.py

```diff
@@ -150,6 +150,56 @@ def test_module_getattr_errors(self):
         if 'test.test_module.bad_getattr2' in sys.modules:
             del sys.modules['test.test_module.bad_getattr2']
 
+    def test_module_setattr(self):
+        import test.test_module.good_setattr as gsa
+        self.assertEqual(gsa.foo, 1)
+        with self.assertRaises(AttributeError):
+            gsa.bar
+        with self.assertRaisesRegex(AttributeError, "Read-only attribute"):
+            gsa.foo = 2
+        gsa.bar = 3
+        self.assertEqual(gsa.bar, 3)
+        del sys.modules['test.test_module.good_setattr']
+
+    def test_module_setattr_errors(self):
+        import test.test_module.bad_setattr as bsa
+        from test.test_module import bad_setattr2
+        self.assertEqual(bsa.foo, 1)
+        self.assertEqual(bad_setattr2.foo, 1)
+        with self.assertRaises(TypeError):
+            bsa.foo = 2
+        with self.assertRaises(TypeError):
+            bad_setattr2.foo = 2
+        del sys.modules['test.test_module.bad_setattr']
+        if 'test.test_module.bad_setattr2' in sys.modules:
+            del sys.modules['test.test_module.bad_setattr2']
+
+    def test_module_delattr(self):
+        import test.test_module.good_delattr as gda
+        self.assertEqual(gda.foo, 1)
+        with self.assertRaises(AttributeError):
+            gda.bar
+        with self.assertRaisesRegex(AttributeError, "Read-only attribute"):
+            del gda.foo
+        gda.bar = 3
+        self.assertEqual(gda.bar, 3)
+        del gda.bar
+        with self.assertRaises(AttributeError):
+            gda.bar
```

Comment:
```suggestion
            gda.bar
            del sys.modules['test.test_module.good_delattr']
```
The part, missing in the last review.